extensions/functions_datetime.yaml

```diff
@@ -104,6 +117,9 @@ scalar_functions:
             value: timestamp_tz
           - name: y
             value: interval_year
+          - name: timezone
```

Comment:
I'm inclined to suggest that we just make a new variation that has timezone rather than change the existing here.

Reply:
The reason we added it is because, as @rok noted, adding a month/year interval to an instant is ill-defined without timezone information. For example, an instant may be February 28th in one timezone and March 1st in another, so to add a month you'd have to add either 28 or 31 days.

That being said, we could still keep the original one with a note stating that it assumes UTC.

Reply:
> That being said, we could still keep the original one with a note stating that it assumes UTC.

Yes, that is what I was requesting.

Reply:
If someone is adding 3 months to an instant, what makes you think that they want to do it in UTC? I don't think defaulting to UTC helps anyone.

Reply:
@jvanstraten I pretty much agree. As you may know, I am one of these extremists who believes that instant, local time, and time-with-time-zone should be kept strictly separate, and therefore I would be inclined to remove the original method.

Reply:
I'm inclined to agree with @julianhyde .  I'd rather only add new functions and implementations when needed and not for the sake of convenience overloads (I also agree that the convenience here is rather dubious).  Otherwise our set of functions gets unwieldy.

There will be a time when we are forced to leave functions around for deprecation reasons but I don't think we are there yet.

@jacques-n are you -1 on removing the old function?  Or was that simply a recommendation?